### ISSUE TITLE
No box is displayed for screen reader when reading widget

### ISSUE THREAD

**Issue by wanchao-xu** [flutter-tizen/engine#261]
I developed the accessibility for flutter-tizen, refer to https://github.com/flutter-tizen/engine/pull/254

There is a box displayed for the widget which is reading.

![m-1215-1-2022-03-18-101409](https://user-images.githubusercontent.com/32251018/158924418-82768ef5-fc5e-434d-aeee-841de6c3dcd5.png)

On android/ios, this box is provied by platform accessibility, the flutter just send the accessibility event to platform, then the box is displayed and widget information is reading.
Tizen platform doesn't provide this box, for example, when screen reader is reading efl widget, screen reader send event to notify efl widget, then efl widget show the box. So when the widget of flutter-tizen is reading, the widget should provide a box.

**Comment by bwikbs** [comment]
I guess.. 
Currently we don't have any parts that use chromium-efl. what does it mean?

**Comment by wanchao-xu** [comment]
flutter-tizen doesn't use chromium-efl, I mean if modify screen reader to provide box, the efl/chromium-efl should be changed not display box.

**Comment by bwikbs** [comment]
I'm sorry...but I can not understand it well. 
Can you explain a little more? I can't imagine know how screen leader draws the box...

**Comment by wanchao-xu** [comment]
Screen reader will read tapped view after accessibility enabled, and a box should be drawn on tapped view at the same time.
The box is drawn by android/ios platform for flutter app, but it's not supported on tizen platform.

If a app is developed by efl on tizen platform, when screen reader is reading the tapped view, this box is drawn by efl view. Screen reader just send a event to efl view, and efl draw this box. If the app is web(displayed by chromium-efl), chromium-efl also implemented the box as efl. So the flutter-tizen should draw the box by itself.

I think the box can be drawn by two ways:
1. Modify flutter-tizen to draw the box, this way is like efl, we should modify the flutter widget(dart) to provide the box.
2. Tizen platform draw this box, this way like android/ios, so I consider whether the screen reader draw this box.

**Comment by bwikbs** [comment]
Thank you for a detailed explanation. :clap: 
There's more questions. 
How draws other platform of flutter do this?
I mean.. Do you know which API is used for this at android?

**Comment by wanchao-xu** [comment]
Android use `AccessibilityNodeProvider` to connect flutter accessibility to system accessibility.
Please refer to https://tool.oschina.net/uploads/apidocs/android/reference/android/view/accessibility/AccessibilityNodeProvider.html

```
/**
 * Bridge between Android's OS accessibility system and Flutter's accessibility system.
 *
 * <p>An {@code AccessibilityBridge} requires:
 *
 * <ul>
 *   <li>A real Android {@link View}, called the {@link #rootAccessibilityView}, which contains a
 *       Flutter UI. The {@link #rootAccessibilityView} is required at the time of {@code
 *       AccessibilityBridge}'s instantiation and is held for the duration of {@code
 *       AccessibilityBridge}'s lifespan. {@code AccessibilityBridge} invokes various accessibility
 *       methods on the {@link #rootAccessibilityView}, e.g., {@link
 *       View#onInitializeAccessibilityNodeInfo(AccessibilityNodeInfo)}. The {@link
 *       #rootAccessibilityView} is expected to notify the {@code AccessibilityBridge} of relevant
 *       interactions: {@link #onAccessibilityHoverEvent(MotionEvent)}, {@link #reset()}, {@link
 *       #updateSemantics(ByteBuffer, String[], ByteBuffer[])}, and {@link
 *       #updateCustomAccessibilityActions(ByteBuffer, String[])}
 *   <li>An {@link AccessibilityChannel} that is connected to the running Flutter app.
 *   <li>Android's {@link AccessibilityManager} to query and listen for accessibility settings.
 *   <li>Android's {@link ContentResolver} to listen for changes to system animation settings.
 * </ul>
 *
 * The {@code AccessibilityBridge} causes Android to treat Flutter {@code SemanticsNode}s as if they
 * were accessible Android {@link View}s. Accessibility requests may be sent from a Flutter widget
 * to the Android OS, as if it were an Android {@link View}, and accessibility events may be
 * consumed by a Flutter widget, as if it were an Android {@link View}. {@code AccessibilityBridge}
 * refers to Flutter's accessible widgets as "virtual views" and identifies them with "virtual view
 * IDs".
 */
public class AccessibilityBridge extends AccessibilityNodeProvider {
```

**Comment by bwikbs** [comment]
Thanks a lot, I learned a lot about accessibility implementation. 👍 
(In my personal opinion) It seems to be a more `tizen-way` to draw it in the engine like efl widget or chromium-efl.
Besides, it is not easy to modify the behavior of efl wiget or chromium-efl too. :smile:
